File: packages/api/pixl_api/schemas/agents.py
"""Pydantic response models for agent endpoints."""

from __future__ import annotations

from pydantic import BaseModel


class AgentResponse(BaseModel):
    """Summary of a configured agent."""

    name: str
    description: str
    effective_model: str = "sonnet"
    default_model: str = "sonnet"
    has_override: bool = False
    override_model: str | None = None
    tools: list[str] = []
    max_turns: int = 50


class ModelResponse(BaseModel):
    """Available LLM model entry."""

    id: str
    provider: str
    description: str | None = None

File: packages/console/src/types/agents.ts
export interface AgentInfo {
  name: string;
  description: string;
  effective_model: string;
  default_model: string;
  has_override: boolean;
  override_model: string | null;
}

export interface UpdateAgentModelRequest {
  model: string | null; // null = reset to default
}

export interface ClassificationModelInfo {
  effective_model: string;
  default_model: string;
  has_override: boolean;
  override_model: string | null;
}

export interface UpdateClassificationModelRequest {
  model: string | null; // null = reset to default
}

export interface SessionReportModelInfo {
  effective_model: string;
  default_model: string;
  has_override: boolean;
  override_model: string | null;
}

export interface UpdateSessionReportModelRequest {
  model: string | null; // null = reset to default
}
